### ISSUE TITLE
vergen's self-tests fail when not run from a git repo

### ISSUE THREAD

**Issue by dkg** [rustyhorde/vergen#140]
If i unpack the vergen 6.0.2 crate, without putting it in a git repo, and i run `cargo test`, i get a bunch of failures:

```
failures:
    gen::test::build_all_kind
    gen::test::build_date_and_time
    gen::test::build_date_only
    gen::test::build_local_timezone
    gen::test::build_time_only
    gen::test::contains_build_output
    gen::test::contains_cargo_output
    gen::test::contains_git_output
    gen::test::contains_no_build_output
    gen::test::contains_no_cargo_output
    gen::test::contains_no_rustc_output
    gen::test::contains_no_sysinfo_output
    gen::test::contains_rustc_output
    gen::test::contains_sysinfo_output
    gen::test::describe
    gen::test::describe_falls_back
    gen::test::detached_head
    gen::test::gen_works
    gen::test::git_all_kind
    gen::test::git_date_and_time
    gen::test::git_date_only
    gen::test::git_dirty
    gen::test::git_local_timezone
    gen::test::git_time_only
    gen::test::semver_kind
    gen::test::sha_kind
    gen::test::toggle_works
    gen::test::vergen_works

test result: FAILED. 48 passed; 28 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.95s
```

I suspect this is related to #124, only it applies to vergen's own self tests when they are not run from the crate.

I believe this is the cause of regressions in [debian's CI](https://ci.debian.net/packages/r/rust-vergen/testing/amd64/) which runs tests on the unpacked crate, rather than within a git repo.

Should we just mark the failing tests as flaky when run not in a git repo?  or is there some way that we should patch or improve them?

**Comment by CraZySacX** [comment]
I'll be adding in support for this use case as part of #141.

**Comment by CraZySacX** [comment]
#146 should address this issue.  Feel free to comment on that PR if you feel I've missed anything.   I may back port some of the new features to older versions as well.

**Comment by CraZySacX** [comment]
This should be resolved with the release of [`8.0.0-beta.0`](https://crates.io/crates/vergen/8.0.0-beta.0).   I also plan to back port the relevant fixes to version 7 and 6 over the next month as well.
